src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -757,12 +771,10 @@ private void AddExecutablePathCommand(IList<LaunchCommand> commands)
 
         private void DetermineAndAddExecutablePathCommand(IList<LaunchCommand> commands)
         {
-            // TODO: rajkumar42, make it robust, either use the liblinux api or fallback shell commands available in other distros.
-            // TODO: rajkumar42, -target-attach can fail with gdb if ptrace>=1, elevate permission with liblinux and retry, display error on second failure.
-            // TODO: rajkumar42, connecting to OSX via SSH doesn't work yet.
+            // TODO: rajkumar42, connecting to OSX via SSH doesn't work yet. Show error after connection manager dialog gets dismissed.
 
             // Runs a shell command to get the full path of the exe.
-            string absoluteExePath = string.Format(CultureInfo.InvariantCulture, @"shell readlink -f /proc/{0}/exe", _launchOptions.ProcessId);
+            string absoluteExePath = System.FormattableString.Invariant($"shell lsof -p {_launchOptions.ProcessId} | awk '$4 == \"txt\" {{ print $9 }}'|awk 'NR==1 {{print $1}}'");
```

Comment:
As discussed, add a comment about what the shell script does (e.g. that you are assuming the first open file will be the executable itself)